socs/agents/hwp_pmx/agent.py

```diff
@@ -52,83 +161,62 @@ def __init__(self, agent, ip, port, f_sample=1, supervisor_id=None,
 
         self.agent.register_feed('rotation_action', record=True)
 
-    @ocs_agent.param('auto_acquire', default=False, type=bool)
-    def init_connection(self, session, params=None):
-        """init_connection(auto_acquire=False)
-        **Task** - Initialize connection to PMX
-
-        Parameters:
-            auto_acquire (bool, optional): Default is False. Starts data
-                acquisition after initialization if True.
-        """
-        if self._initialized:
-            self.log.info("Connection already initialized. Returning...")
-            return True, "Connection already initialized"
-
-        with self.lock.acquire_timeout(0, job='init_connection') as acquired:
-            if not acquired:
-                self.log.warn(
-                    'Could not run init_connection because {} is already running'.format(self.lock.job))
-                return False, 'Could not acquire lock'
-
-            try:
-                self.dev = pmx.PMX(ip=self.ip, port=self.port)
-                self.log.info('Connected to PMX Kikusui')
-            except BrokenPipeError:
-                self.log.error('Could not establish connection to PMX Kikusui')
-                reactor.callFromThread(reactor.stop)
-                return False, 'Unable to connect to PMX Kikusui'
-
-        if params['auto_acquire']:
-            self.agent.start('acq')
-
-        return True, 'Connection to PMX established'
-
     def set_on(self, session, params):
         """set_on()
         **Task** - Turn on the PMX Kikusui.
         """
-        with self.lock.acquire_timeout(3, job='set_on') as acquired:
-            if not acquired:
-                self.log.warn(
-                    'Could not set on because {} is already running'.format(self.lock.job))
-                return False, 'Could not acquire lock'
-
-            if self.shutdown_mode:
-                return False, "Shutdown mode is in effect"
+        if self.shutdown_mode:
+            return False, "Shutdown mode is in effect"
 
-            self.dev.turn_on()
+        action = Actions.SetOn(**params)
+        self.action_queue.put(action)
```

Comment:
Hi Kyohei, in the lockless agents, all tasks should be structured something like this:
```python3
@defer.inlineCallbacks
  def set_on(self, session, params):
    action = Actions.SetOn(**params)
    self.action_queue.put(action)
    session.data = yield action.deferred
    return True, f"Action {action} finished successfully"
```

I'd move the shutdown-mode check to either the `_process_actions` function, or to the action's `process` function, in which case it should be passed in along with the module.

They also need to be registered with `blocking=False`, like
```python3
agent.register_task('set_on', PMX.set_on, blocking=False)
```